dhis2/dhis2-core
# feat: password validation regexp in settings

## Summary
This introduces a new system setting called `passwordValidationPattern`, this is a Regexp version of the internal password rule set, to be used in clients, most important the standard web UI.
Since the password validation is configurable in the back-end, (currently only max and min password length), the rule aka. Regexp needs to be provided dynamically from the back-end.
This new property along with the min and max length properties are also exposed in the `LoginConfigController`, for easy access in non logged in environments, like the signup page.

### Fix
This PR also fixes an issue with the current back-end password rules not checking for a lower-case character present, this is enforced in the UI, but not in the back-end. To align this, a new password rule was added, called `LowerCasePatternValidationRule.java`. Since this rule was expected and always enforced in the UI, I see this as safe to add.

### Automatic tests
SystemSettingsTest.java
UserTest.java
LoginConfigControllerTest.java
UserAccountControllerTest.java
### Manual test
1. Call API endpoint /systemSettings
2. Observe there is a `passwordValidationPattern` property present and that the value is: `^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[\W_])[A-Za-z\d\W_]{8,72}$`
3. Call API endpoint /loginConfig
4. Observe the same property is present along with `minPasswordLength` and `maxPasswordLength`

#### JIRA: [DHIS2-18574](https://dhis2.atlassian.net/browse/DHIS2-18574)

[DHIS2-18574]: https://dhis2.atlassian.net/browse/DHIS2-18574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
